### PR TITLE
TD-2593 - Create React UI Tabbed Panel component

### PR DESCRIPTION
**Pull request by syedsalehinipg**
Closes TD - [TD-2593](https://sce.myjetbrains.com/youtrack/issue/TD-2593/Create-React-UI-Tabbed-Panel-component)
Contributes to [TD-2030](https://sce.myjetbrains.com/youtrack/issue/TD-2030/Migrating-VIRTO.DATA-to-Remix)

## Changes

This PR introduces the TabPanel component, designed to render tabbed interfaces within the application. 

- **New Components**: Added the TabPanel component along with its type definitions and a Storybook story.
- **Storybook Updates**: Created a new Storybook story under Layout/TabPanel to demonstrate the component's functionality and provide examples of its usage.

**Features:**

-   Dynamic tab generation based on child elements.
-   External control of the active tab through a prop for greater flexibility.

**Usage:**

The TabPanel component can be used in any part of the application that requires tabbed content. It accepts any number of children, each representing a tab with its label provided via a `data-label` attribute. The active prop can be used to programmatically control the active tab.

```
<TabPanel active={activeTabIndex}>
  <Box data-label="Tab 1">Content for Tab 1</Box>
  <Box data-label="Tab 2">Content for Tab 2</Box>
  // Add more tabs as needed
</TabPanel>
```

### **Other changes in this PR:**

**TreeViewList** 
  - Modified logic to only show `No search results.` message when there is data but the filtered data is zero. Previously it was showing if even there was no data.
  - Show `No data is available` message when there is no data passed into the component

## UI/UX

_Tab 1_
image

_Tab 2_
image

_TreeViewList (No data):_
image

## Testing notes

Navigate to the Storybook deployment for this PR and review the TabPanel component under Layout/TabPanel. Test the functionality of switching tabs and observe the content changes accordingly.

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
~~- [ ] Branch has been run in docker.~~
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/TabPanel/TabPanel.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { Meta, StoryFn } from "@storybook/react";
+
+import Box from "@mui/material/Box";
+import React from "react";
+import TabPanel from "./TabPanel";
+import { TabPanelProps } from "./TabPanel.types";
+import { TextField } from "@mui/material";
```
Comment (m4manjesh) [inline src/TabPanel/TabPanel.stories.tsx:7]: Can combine this import with Box

File: src/TabPanel/TabPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  // handle tab change
+  const handleChange = (event: React.SyntheticEvent, tab: number) => {
+    setActiveTab(tab);
+  };
+
+  return (
+    <Box>
+      <Tabs
+        value={activeTab}
+        onChange={handleChange}
```
Comment (m4manjesh) [inline src/TabPanel/TabPanel.tsx:32]: Should there be a callback on tab change with clicked tab index so then the app can use it?

**Review by m4manjesh — CHANGES_REQUESTED** [review]
Overall looks good. Just 2 things, see comments.

File: src/TabPanel/TabPanel.tsx (hunk 2 above)
Comment (syedsalehinipg) [reply src/TabPanel/TabPanel.tsx:32]: I didn't think there needs to be, the tab change will just change the view, there isn't anything the parent will do with just a tab change, and anything that happens within that tab will be dealt with within the embed children. I couldn't think of a reason why we need it, if you do I can still add it?

Comment (m4manjesh) [reply src/TabPanel/TabPanel.tsx:32]: As dicussed please add the callback then app will be able to use the index for various purpose , such as to keep the tabe active on reload.

**Comment by syedsalehinipg** [comment]
Changes made, please also see updated description which includes other changes.

**Review by m4manjesh — APPROVED** [review]
LGTM

**Pull request merged**